src/supermarket/app/views/api/v1/cookbook_versions/_cookbook_version.json.jbuilder

```diff
@@ -4,18 +4,15 @@ json.version cookbook_version.version
 json.average_rating nil
 json.cookbook api_v1_cookbook_url(cookbook)
 json.file api_v1_cookbook_version_download_url(cookbook_version.cookbook, cookbook_version)
+
 if !@cookbook_version_metrics.nil? && @cookbook_version_metrics.any?
-  json.quality do
-    json.foodcritic do
-      json.failed foodcritic_metric_result(cookbook_version).failure
-      json.feedback foodcritic_metric_result(cookbook_version).feedback
-    end
-    json.collaborator do
-      json.failed collaborator_num_metric_result(cookbook_version).failure
-      json.feedback collaborator_num_metric_result(cookbook_version).feedback
-    end
+  json.quality_metrics @cookbook_version_metrics do |metric|
```

Comment:
I think the iteration done by the builder thingie used here alone will take care of the `!nil?` and `any?` case. If `@cookbook_version_metrics` is `nil` or `[]`, the iteration will be a no-op.

Whether that is correct or not, I suggest the `if` above also include (or only be) a feature flag check for fieri being enabled.